clients/admin-ui/src/features/common/UpgradeModal.tsx

```diff
@@ -0,0 +1,40 @@
+import { useRouter } from "next/router";
+
+import ConfirmationModal from "~/features/common/ConfirmationModal";
+
+type Props = {
+  nonUpgradeLink: string;
+  upgradeLink: string;
+  isOpen: boolean;
+  isSystem: boolean;
+  onClose: () => void;
+};
+
+export const UpgradeModal = ({
+  nonUpgradeLink,
+  upgradeLink,
+  isOpen,
+  isSystem,
+  onClose,
+}: Props) => {
+  const router = useRouter();
+  const templateWord = isSystem ? "system" : "vendor";
+
+  return (
+    <ConfirmationModal
+      isOpen={isOpen}
+      onClose={onClose}
+      onCancel={() => {
+        router.push(nonUpgradeLink);
+      }}
+      isCentered
+      title={`Upgrade to add multiple ${templateWord}s at once!`}
+      message={`You will need to upgrade your subscription to add multiple ${templateWord}s at once. In the meantime you can continue here to add indivual ${templateWord}s manually.`}
```

Comment:
is there a case we wouldn't want to show this upgrade modal? like if a user is already subscribed to... something? I guess if the dictionary is available?

Reply:
Yes, that's correct. I handled that in the `AddSystem` component :slightly_smiling_face: 

https://github.com/ethyca/fides/blob/aecf560ad52d4f3b28da9c21f4b61ce114e68cbc/clients/admin-ui/src/features/config-wizard/AddSystem.tsx#L91-L98